pkg/awsutils/awsutils.go

```diff
@@ -315,8 +331,27 @@ func (i instrumentedIMDS) GetMetadataWithContext(ctx context.Context, p string)
 	return result, nil
 }
 
+func (cache *EC2InstanceMetadataCache) GetNetworkInterfacesWithContext(ctx context.Context, eniIds []*string) (*ec2.DescribeNetworkInterfacesOutput, error) {
```

Comment:
1. start with lower case for private functions. seems `GetNetworkInterfacesWithContext` is not exposed.
2. it's better to use `eniIds []string`. avoid use pointers when unnecessary.

Reply:
Input to the DescribeNetworkInterface func is array of pointers hence I am passing it to avoid multiple conversions - `input := &ec2.DescribeNetworkInterfacesInput{NetworkInterfaceIds: eniIds}`. Since this is not exposed to any other APIs, is it fine to keep it as pointers?

```

    // One or more network interface IDs.
    //
    // Default: Describes all your network interfaces.
    NetworkInterfaceIds []*string `locationName:"NetworkInterfaceId" locationNameList:"item" type:"list"`
```